### PR TITLE
Feature/api/mrxn23 97 delete cost surface [MRXN23-97]

### PR DESCRIPTION
**Pull request by KevSanchez**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

**Comment by KevSanchez** [comment]
Part of the GET endpoints functionality is in this PR, as it was shared on the CostSurfaceService. Thought it was not worth the hassle to rearrange everything, when the GETs PR is going to be merged on top of this one too.

File: api/apps/api/src/modules/cost-surface/delete-cost-surface/delete-cost-surface.handler.spec.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+it('fails to delete a project ', async () => {
+  const projectId = fixtures.GivenProjectExits();
+  fixtures.GivenDeleteOperationFails();
+
+  const result = await fixtures.WhenACostSurfaceIsDeleted(projectId);
+
+  fixtures.ThenDeleteCostSurfaceFailedIsReturned(result);
+  await fixtures.ThenCostSurfaceIsNotDeleted(projectId);
+  fixtures.ThenNoEventIsEmitted();
+});
```
Comment (hotzevzl) [inline api/apps/api/src/modules/cost-surface/delete-cost-surface/delete-cost-surface.handler.spec.ts:43]: very minor (feel free to keep as is):
- I guess the test description is a leftover from copy/paste - it should likely be like `won't delete cost data when deleting the relevant cost surface fails` maybe?
- `GivenDeleteOperationFails()` - I had to follow the code to understand what this does - maybe `GivenCostSurfaceDeletionOperationWillFail()` or something like this?
- `ThenNoEventIsEmitted()` it is technically exactly what the function is checking, though if we want to make what is happening clearer, maybe we should call this `ThenNoCostSurfaceDeletedEventIsEmitted()`

**Review by hotzevzl — APPROVED** [review]

File: api/apps/api/src/modules/cost-surface/delete-cost-surface/delete-cost-surface.handler.spec.ts (hunk 1 above)
Comment (KevSanchez) [reply api/apps/api/src/modules/cost-surface/delete-cost-surface/delete-cost-surface.handler.spec.ts:43]: First one seems like the description might have been incomplete from the get go, and I had forgotten about it.
The other two, I thought it was implied enough from the context, but I see what you mean and I  do agree.
These changes seem minor enough that I could sneak them in another PR once this one is merged, don't they?

Comment (hotzevzl) [reply api/apps/api/src/modules/cost-surface/delete-cost-surface/delete-cost-surface.handler.spec.ts:43]: yes, let's just get this merged, thanks!

**Pull request merged**